src/pal/src/numa/numa.cpp

```diff
@@ -140,71 +142,91 @@ BOOL
 NUMASupportInitialize()
 {
 #if HAVE_NUMA_H
-    if (numa_available() != -1)
+    numaHandle = dlopen("libnuma.so", RTLD_LAZY);
+    if (numaHandle == 0)
     {
-        struct bitmask *mask = numa_allocate_cpumask();
-        int numaNodesCount = numa_max_node() + 1;
-
-        g_possibleCpuCount = numa_num_possible_cpus();
-        g_cpuCount = 0;
-        g_groupCount = 0;
+        numaHandle = dlopen("libnuma.so.1", RTLD_LAZY);
+    }
+    else if (numaHandle != 0)
+    {
+#define PER_FUNCTION_BLOCK(fn) \
+    fn##_ptr = (decltype(fn)*)dlsym(numaHandle, fn); \
+    if (fn##_ptr == NULL) { fprintf(stderr, "Cannot get symbol " #fn " from libnuma\n"); abort(); }
+FOR_ALL_NUMA_FUNCTIONS
+#undef PER_FUNCTION_BLOCK
 
-        for (int i = 0; i < numaNodesCount; i++)
+        if ((*numa_available)() != -1)
```

Comment:
This should be just numa_available(), there is no need to dereference. The same for all other functions.